### DISCUSSION TITLE
I'm trying to figure out if Saleor is good for my ecommerce project (multi-frontend, translations, seo, etc)

### DISCUSSION THREAD

**Discussion by OdifYltsaeb** [saleor/saleor#12287] [Q&A] [answered]
Hey. I have a number of features that are required, and for most of them, there is at least some ( but not comprehensive ) information on how to achieve them.

1. **Support for multi frontends on the same backend**. So there are channels, right? But I can't set categories per channel - from exploring the demo dashboard. I can set some private and public metadata on channels, though. Do you know if that is how I would do that? Or perhaps create it by creating multiple "root" categories as the first level of categories in the admin and then set in each front-ends variable the Id where these frontends categories are?
2. **Translations**. So there's this: https://docs.saleor.io/docs/3.x/developer/extending/api/i18n. So I'm familiar with python, and I've used Django a fair bit, but it's unclear where I'm supposed to put this file. Is this only going to work if I'm either cloning the Saleor repository or creating my Django project with Saleor installed in one of the subfolders?
3. **Per channel SEO** - and per language product SEO. So to get SEO per language, do I need to translate some kind of SEO models too? And do it for both categories and products?
4. **dashboard changes** - to make all the above work neatly. There seems to be no language switch on the product view in the dashboard. How can I add easy ways to switch the language I edit on the dashboard? Can this be done with the apps (https://docs.saleor.io/docs/3.x/developer/extending/apps/key-concepts) by creating some tiny Next projects, or do I have to clone and extend the dashboard for this too?

Would appreciate it a lot if someone could give me some comprehensive answers to those questions and point me towards the right examples, blog posts or code pieces. Or perhaps if all those things can't be done or would be too hacky to do, then please do suggest alternative e-commerce engines.

All the best,
Alan

**Comment by korycins** [comment, marked as answer]
Hey @OdifYltsaeb 
Consider trying [Saleor Cloud](https://saleor.io/pricing/). The developer account is free, so you can create your own Saleor instance and test everything you would like (I recommend using a fictional database).

Ad1. Yes, exactly. Channels allow you to have a single backend to support multiple frontends. You can define which products should be available on a specific channel. If you want to have specific categories shown only on a specific channel, you can use meta or category filters via `slugs` or `ids` ([https://docs.saleor.io/docs/3.x/api-reference/queries/categories#code-style-fontweight-normal-categoriesbfilterbcodecategoryfilterinput-](https://docs.saleor.io/docs/3.x/api-reference/queries/categories#code-style-fontweight-normal-categoriesbfilterbcodecategoryfilterinput-)).

Ad2. What kind of translation are you referring to? If you mean the language of the dashboard for a specific staff user, you can change it in the account settings. If you mean the translation of products/categories, you can check the `translations` section on the dashboard side. You should be able to provide a translation for each object available in the storefront.

Ad3. You can set SEO on the product/category view. Then, in the translation section on the dashboard side, you can provide a translated SEO.

Ad4. The dashboard language can be set in the account settings. In the product/category view, provide the name and description in the main language, then you can provide a translation for any specific language you would like to support.

------

We generally do not recommend forking the repository and modifying the code on your own. The merge process of your changes and a newer version of Saleor can be very time-consuming. If you want to customize a dashboard or extend it by providing specific functionalities, we recommend using Saleor Apps. You can find more information on the key concepts of these apps at [https://docs.saleor.io/docs/3.x/developer/extending/apps/key-concepts](https://docs.saleor.io/docs/3.x/developer/extending/apps/key-concepts).

To check the available apps, go to the Dashboard → Apps section. There is also an app template that you can fork and modify on your own.

**Reply by OdifYltsaeb** [reply]
Hey @korycins Thanks for the answers and your time!

I might indeed try Saleor cloud, just to test the features. Thanks

1. Awesome explanation. Thank you. That seems a bit hacky, but it might work just fine. But what about collections? What's the reason for having them?
2. I was talking about Product and category translations. Your docs (https://docs.saleor.io/docs/3.x/developer/extending/api/i18n) made it look like it can only be done via code changes and extending the codebase somehow. Your answer, on the other hand, tells me it should work out of the box - and it does on your demo dashboard. So which it is, I'm unsure.
3. I Found it on the demo dash. Thank you!
4. I could not find the translations - I did not look deep enough. I thought they were only meant for translating some kind of project strings like Django-rosetta used to.

Thank you

I do have follow-up questions for you. Perhaps I can schedule a call with you or someone else to ask those questions? I'm trying to pick the best possible tech stack, and e-commerce solution for my project and I'd like to make sure I can use Saleor instead of WooCommerce or Magento. I have questions about SSO, Taxes (including 0 VIES tax), Extending APIs (or If there's even a need for that) and so on.

**Reply by korycins** [reply]
# 

Ad1. Category stores a product with a given type, where you could have a collection that contain different type of products but related by some common attribue.  Collections can be also filtered by `ids` or `slugs`.

Ad2. The chapter you are sharing describes the integration of new translations from a code perspective. Here is a chapter related to translations: [https://docs.saleor.io/docs/3.x/api-usage/i18n](https://docs.saleor.io/docs/3.x/api-usage/i18n)

Ad4. We are using Transifex ([https://docs.saleor.io/docs/3.x/developer/extending/api/i18n#translation](https://docs.saleor.io/docs/3.x/developer/extending/api/i18n#translation)) to translate the dashboard panel. However, objects such as products or categories can be translated directly from the dashboard.

> I do have follow-up questions for you. Perhaps I can schedule a call with you or someone else to ask those questions? I'm trying to pick the best possible tech stack, and e-commerce solution for my project and I'd like to make sure I can use Saleor instead of WooCommerce or Magento. I have questions about SSO, Taxes (including 0 VIES tax), Extending APIs (or If there's even a need for that) and so on.

You can contact us by sending an email to [hello@saleor.io](mailto:hello@saleor.io)

**Reply by OdifYltsaeb** [reply]
Awesome! Thank you!

**Comment by michalmirumee** [comment]
Hi @OdifYltsaeb! If you still have any problems with Saleor, I would love to have a quick chat - we are devs working with Saleor on a daily basis. A call maybe?

**Reply by OdifYltsaeb** [reply]
A call would be nice, yes. I would appreciate it a lot.  Sorry for responding almost a month later.  How should we do it? Message me privately and let's set it up?
